src/facade/dragonfly_connection.cc

```diff
@@ -192,6 +203,18 @@ void Connection::RequestDeleter::operator()(Request* req) const {
   mi_free(req);
 }
 
+void Connection::Request::Emplace(RespVec args, size_t capacity) {
+  PipelineMsg* msg = get_if<PipelineMsg>(&payload);
+  if (msg) {
+    if (msg->storage.size() < capacity) {
+      msg->storage.resize(capacity);
+    }
+  } else {
+    payload = PipelineMsg{args.size(), capacity};
+  }
+  SetArgs(args);
```

Comment:
You pass arg on by value here and in other places, but its actually a vector, not a slice

Reply:
Good catch 👍🏼